tests/test_filterblock.py

```diff
@@ -0,0 +1,41 @@
+# Standard
+from unittest.mock import patch
+import operator
+import unittest
+
+# Third Party
+from datasets import Dataset, Features, Value
+
+# First Party
+from instructlab.sdg.filterblock import FilterByValueBlock
+
+
+class TestFilterByValueBlock(unittest.TestCase):
+    def setUp(self):
+        self.block = FilterByValueBlock(
+            filter_column="age",
+            filter_value=30,
+            operation=operator.eq,
+            convert_dtype=int,
+        )
+        self.dataset = Dataset.from_dict(
+            {"age": ["25", "30", "35", "forty", "45"]},
+            features=Features({"age": Value("string")}),
+        )
+
+    def test_convert_dtype_success(self):
+        sample = {"age": "25"}
+        converted_sample = self.block._convert_dtype(sample)
```

Comment:
FWIW I think I would have been inclined to not test the internal `_convert_dtype()` but instead exercise it just through Block.generate()

Reply:
Makes sense. The last case at the end does that. I could just remove these tests of the internal thing. It’d be less code but we still have the same test coverage.

Reply:
I removed the tests of the internal method as suggested, since they don't actually add additional test coverage